client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -556,6 +562,28 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
     commitManager.handleGetReducerFileGroup(context, shuffleId)
   }
 
+  private def handleGetShuffleId(
+      context: RpcCallContext,
+      appShuffleId: Int,
+      maxAttemptNum: Int,
+      attemptId: Int,
+      isShuffleWriter: Boolean): Unit = {
+
+    val newShuffleId = appShuffleId * maxAttemptNum + attemptId
+    val shuffleIds = shuffleIdMapping.computeIfAbsent(appShuffleId, new function.Function[Int, Seq[Int]]() {
+      override def apply(shuffleId: Int): Seq[Int] = Seq.apply(newShuffleId)
+    })
+
+    if (isShuffleWriter && !shuffleIds.contains(newShuffleId)) {
+      logInfo(s"[handleGetShuffleId]" +
+        s"record new shuffleId $newShuffleId with appShuffleId $appShuffleId attemptId $attemptId")
+      shuffleIdMapping.put(appShuffleId, shuffleIds :+ newShuffleId)
+    }
+
+    val shuffleId = shuffleIdMapping.get(appShuffleId).last
```

Comment:
For Shuffle unregister/clean uselessShuffleId 
1. When register a new shuffleId with same appShuffleId we can clear all out-dated info in lifecycleManager, such as shuffleIdMapping(appShuffleId->shuffleId), shuffleId related info...
2. When shuffleId unregister, LifecycleManager alse can clear the useless shuffleId in shuffleIdMapping.

Reply:
shuffle clean up has been added:
1. In LifecycleManager, unregisterShuffle() is invoked when handing fetch failure
2. new function unregisterAppShuffle() is added to LifecycleManager what will be called by SparkShuffleManager.unregisterShuffle() with appShuffleId
3. regarding executor, shuffleIdTracker is added to track and cleanup related shuffle id in shuffleClient